e2etests/testcase_test.go

```diff
@@ -29,11 +28,19 @@ type testCase struct {
 	requiredFeatureFlag features.FeatureFlag
 }
 
-var once sync.Once
+func init() {
+	enabledCases := testCases[:0]
+	for _, tc := range testCases {
+		// Filter out test cases that depend on disabled feature flags.
+		if tc.requiredFeatureFlag != nil && !tc.requiredFeatureFlag.Enabled() {
+			continue
+		}
+		enabledCases = append(enabledCases, tc)
+	}
+	testCases = enabledCases
+}
 
 // testCases defines all the E2E test cases.
-// testCases should not be used directly;
-// instead, users should call getEnabledTestCases().
```

Comment:
Could you please leave a comment on how this might be filtered by feature flags?